src/symmetric_crypto/aes_256_gcm_pure/dem.rs

```diff
@@ -16,15 +17,17 @@ impl Dem for Aes256GcmCrypto {
         additional_data: Option<&[u8]>,
         message: &[u8],
     ) -> Result<Vec<u8>, CryptoCoreError> {
-        if secret_key.len() < Self::Key::LENGTH {
+        if secret_key.len() < <<Self::Key as KeyTrait>::Length as Unsigned>::to_usize() {
             return Err(CryptoCoreError::SizeError {
                 given: secret_key.len(),
-                expected: Self::Key::LENGTH,
+                expected: <<Self::Key as KeyTrait>::Length as Unsigned>::to_usize(),
             });
         }
         // AES GCM includes an authentication method
         // there is no need for parsing a MAC key
-        let key = Self::Key::try_from(&secret_key[..Self::Key::LENGTH])?;
+        let key = Self::Key::try_from(
+            &secret_key[..<<Self::Key as KeyTrait>::Length as Unsigned>::to_usize()],
```

Comment:
maybe use a variable in this function to store `<<Self::Key as KeyTrait>::Length as Unsigned>::to_usize()`?